model-optimizer/extensions/front/kaldi/tdnn_component_replacer.py

```diff
@@ -0,0 +1,104 @@
+"""
+ Copyright (C) 2018-2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from extensions.ops.MatMul import FullyConnected
+from mo.front.common.replacement import FrontReplacementPattern
+from mo.graph.graph import Graph, Node
+from mo.graph.graph import rename_nodes
+from mo.ops.concat import Concat
+from mo.ops.const import Const
+from mo.ops.memoryoffset import MemoryOffset
+
+
+class TdnnComponentReplacer(FrontReplacementPattern):
+    '''
+    Expand TdnnComponent into MemoryOffsets, Concat and FullyConected nodes
+
+    BEFORE:
+                          placeholder
+                              |
+                      TdnnComponent('time_offsets': t1, t2,... tk)
+                              |
+    _______________________________________________________________
+
+    AFTER:
+                          placeholder
+            __________________|___________________________
+           /                  |              \            \
+   MemoryOffset(t1)     MemoryOffset(t2)    ...    MemoryOffset(tk)
+          \_____________ _____|______________/____________/
+                           Concat
+                             |
+                        FullyConnected
+                             |
+    '''
+    enabled = True
+    run_not_recursively = True
+
+    def run_after(self):
+        from extensions.front.restore_ports import RestorePorts
+        return [RestorePorts]
+
+    def run_before(self):
+        from extensions.front.kaldi.memory_offset_adjustment import MemoryOffsetAdjustment
+        return [MemoryOffsetAdjustment]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        for node in graph.get_op_nodes(op='tdnncomponent'):
+            self.replace_tdnn(graph, node)
+
+    def replace_tdnn(self, graph: Graph, tdnn_node: Node):
+        tdnn_name = tdnn_node.soft_get('name', tdnn_node.id)
+
+        concat_node = Concat(graph, {'axis': 1}).create_node()
+        rename_nodes([(tdnn_node, tdnn_name + '/to_be_removed'), (concat_node, tdnn_name)])
+
+        for i, t in enumerate(tdnn_node['time_offsets']):
```

Comment:
nit: rename `i` to something more meaningful like `offset_ind`

Reply:
Done.